design/003-interactive-session/003-interactive-session.md

```diff
@@ -0,0 +1,94 @@
+- Start Date: 2021-12-02
+- Status: Proposed
+
+# Run local interactive sessions
+
+## Summary
+
+Propose a new sub-group command `renku session` that is responsible for
+managing local interactive sessions for a project.
+
+## Motivation
+
+Renkulab.io provides the option to run interactive sessions of a project. Among other things the session
+contains a jupyter notebook.
+
+The motivation behind this RFC is to provide a simple renku CLI command that basically mimics this, i.e. the
+option to start an interactive session of a project locally.
+
+## Design Detail
+
+The proposed sub-commands of `renku session`, namely to run interactive sessions locally, heavily relies on
+[docker](https://www.docker.com/). Docker is one of the most popular container engines, that eases the development
+and deployment of applications, by off-loading the installation of the runtime environment dependencies of the
+application.
+
+In order to start a local interactive session, just as in case of renkulab.io, it is required to provide
+a `Dockerfile`. This is by default provided by `renku init` when creating a renku project, but if the user
+is using a custom project template it might not be present.
+
+### Proposed Commands
+
+#### renku session start
+
+A command to start an interactive session.
+
+By default, the `Dockerfile` present in the given project is going to be used for building the docker image
+that's going to be used for running the interactive session. In case the docker image is not available the
+user is going to be prompted whether the user wants to build the docker image, as in some cases it can be
+time consuming.
+
+##### Detailed Parameter Description
+
+```
+renku session start [OPTIONS]
```

Comment:
If we want to consider the more general case of launching remote sessions as well from the CLI, we might add a `--remote` flag here.

Reply:
while implementing this functionality couple of questions came up:
 - remote notebook sessions have various additional parameters that are in case of local sessions might not have need to be defined (of course they could be), namely the resource based parameters:
```
  "serverOptions": {
    "cpu_request": 0.25,
    "defaultUrl": "/lab",
    "disk_request": "1G",
    "gpu_request": 0,
    "lfs_auto_fetch": false,
    "mem_request": "1G"
  }
``` 
in case of a remote session, i believe these parameters should be exposed to the `docker session start --remote`, so that a user can override the default values (in case of a local session these params are not necessarily needed to be passed to docker, although it's possible). but since by design (or at least atm) the remote starting of a session  should be just a flag of `docker session start`, i believe that they should be exposed.

 * if we do support `--remote` from the CLI then we should support for all the subcommands, i.e. `start, stop, list`.      
     * should we include the remote sessions for `renku workflow ls` by default? or should it be the exact same behaviour like in case of starting a session (i.e. local by default)
     * in case of `docker session stop --all` should be both for local and remote sessions?

Reply:
imho this should not be implemented now, it was just a suggestion to add to the RFC to be implemented in the future, precisely because of the issues you raise, i.e. it requires a design all on its own.